src/protagonist/API/Features/Image/Ingest/AssetProcessor.cs

```diff
@@ -91,7 +92,15 @@ public async Task<ProcessAssetResult> Process(AssetBeforeProcessing assetBeforeP
 
                 counts.CustomerStorage.NumberOfStoredImages++;
             }
-            else if (assetBeforeProcessing.DeliveryChannelsBeforeProcessing.IsNullOrEmpty() && alwaysReingest)
+            else if (assetBeforeProcessing.DeliveryChannelsBeforeProcessing == null && alwaysReingest)
+            {
+                assetBeforeProcessing.DeliveryChannelsBeforeProcessing = assetFromDatabase.ImageDeliveryChannels

```

Comment:
Is this going beyond what `DeliveryChannelsBeforeProcessing` were intended for? They were originally a carrier of "brief" data about delivery-channels to be set (channel + policyName) but this is taking the fully fleshed out data from DB and then reducing it down to the "brief" data and then passing it to the `DeliveryChannelProcessor`, which would then reprocess it (when we already know it?)